client/src/layouts/SignedOutLayout.jsx

```diff
@@ -27,15 +32,42 @@ const useStyles = makeStyles((theme) => ({
 
 const SignedOutLayout = ({ children, isLandingPage }) => {
   const classes = useStyles();
+  const { user, setUser } = useContext(UserContext);
+  const [error, setError] = useState('');
+
+  useEffect(() => {
+    if (!user) {
+      AuthApi.getProfileInfo()
+        .then((response) => {
+          setUser(response.data);
+        })
+        .catch((err) => {
+          if (!err.response || err.response.status !== 401) {
+            setError('An unexpected error occurred while contacting the API.');
+          }
+        });
+    }
+  }, [user, setUser]);
```

Comment:
Just a small thought - does it make most sense to have this here? If someone returns to the application after logging in, it would go into WrappedRoute, and then direct them to SignedOutLayout (as user is not defined yet), and then back to AppLayout once the user is fetched, meaning that the user might be jumping between routes

Instead should we aim to get the user in the App.js and also have a loading state so that we show a loading sign while we get the user as opposed to oscillating between routes?

Reply:
Fixed!

Got confused in this kinda new (for me) context stuff as to where to store/pass state, having to use useEffect should have been a cue that it wasn't quite the right path :P